src/transformers/models/owlvit/modeling_tf_owlvit.py

```diff
@@ -0,0 +1,1730 @@
+            image_class_embeds / tf.norm(tensor=image_class_embeds, ord="euclidean", axis=-1, keepdims=True) + 1e-6
+        )
+        query_embeds = query_embeds / tf.norm(tensor=query_embeds, ord="euclidean", axis=-1, keepdims=True) + 1e-6
+
+        # Get class predictions
+        pred_logits = tf.einsum("...pd,...qd->...pq", image_class_embeds, query_embeds)
+
+        # Apply a learnable shift and scale to logits
+        logit_shift = self.logit_shift(image_embeds)
+        logit_scale = self.logit_scale(image_embeds)
+        logit_scale = self.elu(logit_scale) + 1
+        pred_logits = (pred_logits + logit_shift) * logit_scale
+
+        if query_mask is not None:
+            if len(query_mask.shape) > 1:
+                query_mask = tf.expand_dims(query_mask, axis=-2)
+
+            pred_logits = tf.where(query_mask == 0, -1e6, pred_logits)
+
+        return (pred_logits, image_class_embeds)
+
+
+class TFOwlViTForObjectDetection(TFOwlViTPreTrainedModel):
+    config_class = OwlViTConfig
+
+    def __init__(self, config: OwlViTConfig, *inputs, **kwargs):
+        super().__init__(config, *inputs, **kwargs)
+
+        self.owlvit = TFOwlViTMainLayer(config, name="owlvit")
+        self.class_head = TFOwlViTClassPredictionHead(config, name="class_head")
+        self.box_head = TFOwlViTBoxPredictionHead(config, name="box_head")
+        self.layer_norm = tf.keras.layers.LayerNormalization(
+            epsilon=config.vision_config.layer_norm_eps, name="layernorm"
+        )
+
+    def normalize_grid_corner_coordinates(self, feature_map: Optional[TFModelInputType] = None):
+        # Computes normalized xy corner coordinates from feature_map.
+        if not len(feature_map.shape) == 4:
+            raise ValueError("Expected input shape is [batch_size, num_channels, height, width]")
+
+        height, width = feature_map.shape[1], feature_map.shape[2]
+
+        box_coordinates = np.stack(np.meshgrid(np.arange(1, width + 1), np.arange(1, height + 1)), axis=-1).astype(
+            np.float32
+        )
+        box_coordinates /= np.array([width, height], np.float32)
+
+        # Flatten (h, w, 2) -> (h*w, 2)
+        box_coordinates = box_coordinates.reshape(
+            box_coordinates.shape[0] * box_coordinates.shape[1], box_coordinates.shape[2]
+        )
+        box_coordinates = tf.convert_to_tensor(box_coordinates)
+
+        return box_coordinates
+
+    def compute_box_bias(self, feature_map: tf.Tensor) -> tf.Tensor:
+        # The box center is biased to its position on the feature grid
+        box_coordinates = self.normalize_grid_corner_coordinates(feature_map)
+        box_coordinates = tf.clip_by_value(box_coordinates, 0.0, 1.0)
+
+        # Unnormalize xy
+        box_coord_bias = tf.math.log(box_coordinates + 1e-4) - tf.math.log1p(-box_coordinates + 1e-4)
+
+        # The box size is biased to the patch size
+        box_size = tf.experimental.numpy.full_like(box_coord_bias, 1.0 / feature_map.shape[-2])
```

Comment:
I don't think we officially support `tf.experimental` 😭 
```suggestion
        box_size = tf.ones_like(box_coord_bias) * 1.0 / feature_map.shape[-2]
```

Reply:
Also good to know, I will switch to using `full_like` when it's fully supported by tf